clients/python/dd_image_client.py

```diff
@@ -0,0 +1,316 @@
+import base64
+import time
+import json
+import os
+from dd_client import DD
+import cStringIO
+from PIL import Image
+from PIL import ExifTags
+
+
+from scipy import spatial
+
+
+class DDImage(object):
+
+    mllib = 'caffe'
+
+    def __init__(self, host):
+
+        self.host = host
+        self.sname = ''
+        self.description = ''
+        self.mltype = ''
+        self.extract_layer = ''
+        self.nclasses = 0
+        self.width = self.height = 224
+        self.binarized = False
+        self.GPU = True
+        self.model_repo = ''
+        self.dd = None
+        self.model_name = ''
+        self.batch_size = None
+        self.GOOGLENET_BATCH_SIZE = 40
+        self.RESNET152_BATCH_SIZE = 4
+        self.model_repo
+
+    def init_for_classification(self, model_name, model_repo):
+        self.sname = "_classification"
+        self.description = self.sname + ' for classification'
+        self.mltype = 'supervised'
+        self.nclasses = 1000
+        self.model_name = model_name
+        self.model_repo=model_repo
+        self.dd = DD(self.host)
+        self.dd.set_return_format(self.dd.RETURN_PYTHON)
+
+    def init_for_extraction(self, model_name, model_repo):
+        self.sname = model_name + "_extraction"
+        self.description = self.sname + ' for feature extraction'
+        self.mltype = 'unsupervised'
+        self.model_name = model_name
+        self.model_repo = model_repo
+
+        if model_name == 'resnet152':
+            self.extract_layer = 'fc1000'
+            self.nclasses = 1000
+            self.batch_size = self.RESNET152_BATCH_SIZE
+        elif model_name == 'googlenet':
+            self.extract_layer = 'pool5/7x7_s1'
+            self.nclasses = 1000
+            self.batch_size = self.GOOGLENET_BATCH_SIZE
+        else:
```

Comment:
Templates are just the tip of the iceberg. Basically most of the Caffe model zoo can be used. Also, the number of classes cannot be fixed, most applications finetune these pre-trained models for custom applications. My hunch is that having a generic image client, and adding the table suggested in #130 is the right way to go. Open to suggestions to move faster though, if this is too much work. Note that just an embryo of table is needed, we can help filling it up.